File: reverse-vowels/reverse-vowels.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        # edge case if length of string is 1
        if len(s) == 1:
            return s
        # other string is length of 1 or greater
        else:
            temp_s = list(s)

            # find the vowels and keep track of index of vowels
            vowelIndexes = []
            for i in range(len(s)):
                letter = s[i]
                if letter in 'aeiouAEIOU':
                    vowelIndexes.append(i)

            if len(vowelIndexes) == 1 or len(vowelIndexes) == 0:
                return s

            # swap vowels
            for index in range(len(vowelIndexes) // 2):
                vowelIndex = vowelIndexes[index]
                vowelIndex2 = vowelIndexes[-index-1]

                tempVowel = temp_s[vowelIndex]
                temp_s[vowelIndex] = temp_s[vowelIndex2]
                temp_s[vowelIndex2] = tempVowel

            return ''.join(temp_s)

File: reverse-vowels/reverse-vowels.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    var sArray = [...s]
    const vowelIndexes = []

    for (let i=0; i < sArray.length; i++) {
        var letter = sArray[i]
        if ('AEIOUaeiou'.includes(letter)) {
            vowelIndexes.push(i)
        }
    }

    if (vowelIndexes.length === 1 || vowelIndexes.length === 0) {
        return s
    } 

    for (let i=0; i < parseInt(vowelIndexes.length / 2); i++) {
        var vowelIndex = vowelIndexes[i]
        console.log(-i-1)
        var vowelIndex2 = vowelIndexes[vowelIndexes.length-1-i]

        var tempVowel = sArray[vowelIndex]

        console.log(tempVowel, vowelIndex, vowelIndex2)
        sArray[vowelIndex] = sArray[vowelIndex2]
        sArray[vowelIndex2] = tempVowel 
    }

    return sArray.join('')
};